core/src/main/scala/kafka/coordinator/GroupCoordinator.scala

```diff
@@ -412,11 +419,16 @@ class GroupCoordinator(val brokerId: Int,
           } else if (generationId != group.generationId) {
             responseCallback(offsetMetadata.mapValues(_ => Errors.ILLEGAL_GENERATION.code))
           } else {
-            groupManager.storeOffsets(groupId, memberId, generationId, offsetMetadata, responseCallback)
+            delayedStore = Some(groupManager.prepareStoreOffsets(groupId, memberId, generationId,
+              offsetMetadata, responseCallback))
           }
         }
       }
     }
+
+    // Append the offset commit to the log without holding the group metadata lock
+    if (delayedStore.isDefined)
```

Comment:
Use `delayedStore.foreach` instead of `isDefined` and `get` (one should ideally never use `get`).